File: 3-Archive/Secure-Storage-ObjC/Secure Storage Sample Obj-C/CommonVC.m
/* Copyright (c) 2018 BlackBerry Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

#import "CommonVC.h"
#import "SecureCoreDataVC.h"

@interface CommonVC ()

@end

@implementation CommonVC

//Setup background image
- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIImageView *imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"book"]];
    [self.tableView.layer setCornerRadius:20];
    self.tableView.backgroundColor = [UIColor clearColor];
    
    UIBlurEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleExtraLight];
    UIVisualEffectView *blurEffectView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
    [blurEffectView setFrame:self.view.frame];
    [imageView addSubview:blurEffectView];
    self.tableView.backgroundView = imageView;
}

//Switch the delegate class
-(void)viewWillAppear:(BOOL)animated {
    
    if ([NSStringFromClass([[self parentViewController] classForCoder]) containsString:@"SecureSQL"]) {
        
        SecureCoreDataVC *vc = (SecureCoreDataVC*)[self parentViewController];
        self.tableView.delegate = vc;
        self.tableView.dataSource = vc;
        vc.tableDelegate = self;
        
    } else {
        
        SecureCoreDataVC *vc = (SecureCoreDataVC*)[self parentViewController];
        self.tableView.delegate = vc;
        self.tableView.dataSource = vc;
        vc.tableDelegate = self;
    }
    [self reloadTable];
}

-(void)reloadTable {
    [self.tableView reloadData];
}


@end

File: 3-Archive/Secure-Storage/Secure Storage Sample Swift/CommonVC.swift
/* Copyright (c) 2018 BlackBerry Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

import UIKit

//Protocol to reload table
protocol reloadTable {
    func reloadTable()
}

class CommonVC: UIViewController, reloadTable {
    
    //Protocol function
    func reloadTable() {
        tableView.reloadData()
    }

    @IBOutlet weak var tableView: UITableView!
    
    //Setup background image
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let imageView = UIImageView(image: UIImage.init(named: "book.jpg")!)
        tableView.layer.cornerRadius = 20
        self.tableView.backgroundColor = UIColor.clear
        
        let blurEffect = UIBlurEffect(style: .extraLight)
        let blurEffectView = UIVisualEffectView(effect: blurEffect)
        blurEffectView.frame = self.view.frame
        imageView.addSubview(blurEffectView)
        self.tableView.backgroundView = imageView
    }
    
    //Switch the delegate class
    override func viewWillAppear(_ animated: Bool) {
        tableView.delegate = self.parent as? UITableViewDelegate
        tableView.dataSource = self.parent as? UITableViewDataSource
        
        if NSStringFromClass((self.parent?.classForCoder)!).contains("SecureSQL") {
            (self.parent as! SecureSQLVC).tableDelegate = self
        } else {
            (self.parent as! SecureCoreDataVC).tableDelegate = self
        }
        tableView.reloadData()
    }
}
